drivers/usb/device/usb_dc_sam0.c

```diff
@@ -351,6 +347,8 @@ int usb_dc_ep_configure(const struct usb_dc_ep_cfg_data *const cfg)
 	uint8_t ep = cfg->ep_addr & ~USB_EP_DIR_MASK;
 	UsbDeviceEndpoint *endpoint = &regs->DeviceEndpoint[ep];
 	UsbDeviceDescriptor *desc = &data->descriptors[ep];
+	UsbDeviceDescBank *bank = &desc->DeviceDescBank[for_in ? 1 : 0];
```

Comment:
TBH, I do not like this kind of obfuscation (for_in is a bad name too, `is_in_ep` would probably be clearer). Here it would only be two lines with `if (cfg->ep_addr & USB_EP_DIR_MASK) {}` so you probably do not need `for_in` at all.

Reply:
I'm not worried, so I removed the use of for_in and changed to the corresponding if's.